File: app/src/main/java/com/example/intern_anrdoid_2020/view/QiitaListViewAdapter.java
package com.example.intern_anrdoid_2020.view;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.bumptech.glide.Glide;
import com.example.intern_anrdoid_2020.R;
import com.example.intern_anrdoid_2020.model.response.QiitaArticleResponse;

import java.util.List;

public class QiitaListViewAdapter extends RecyclerView.Adapter<QiitaListViewAdapter.QiitaListViewHolder> {

    private List<QiitaArticleResponse> listData;

    private OnClickItemListener listener;

    public interface OnClickItemListener {
        void onClickQiitaListData(QiitaArticleResponse item);
    }

    public void setOnClickItemListener(OnClickItemListener listener) {
        this.listener = listener;
    }

    public static class QiitaListViewHolder extends RecyclerView.ViewHolder {

        private ImageView ivQiitaImage;
        private TextView tvQiitaTitle;
        private TextView tvQiitaLgtm;

        private QiitaListViewHolder(View itemView) {
            super(itemView);
            ivQiitaImage = itemView.findViewById(R.id.iv_book_image);
            tvQiitaTitle = itemView.findViewById(R.id.tv_book_title);
            tvQiitaLgtm = itemView.findViewById(R.id.tv_book_price_tax);
        }
    }

    public QiitaListViewAdapter(List<QiitaArticleResponse> qiitaListData) {
        this.listData = qiitaListData;
    }

    @NonNull
    @Override
    public QiitaListViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View v = LayoutInflater.from(parent.getContext()).inflate(R.layout.row_qiita_list, parent, false);
        return new QiitaListViewHolder(v);
    }

    @Override
    public void onBindViewHolder(QiitaListViewHolder holder, int position) {

        QiitaArticleResponse qiitaArticleResponse = listData.get(position);

        holder.tvQiitaTitle.setText(qiitaArticleResponse.title);
        holder.tvQiitaLgtm.setText(String.valueOf(qiitaArticleResponse.likesCount));
        Glide.with(holder.ivQiitaImage.getContext()).load(qiitaArticleResponse.user.profileImageUrl).into(holder.ivQiitaImage);
        holder.itemView.setOnClickListener((View v) -> {
            if (position == RecyclerView.NO_POSITION) {
                return;
            }
            if (listener != null) {
                listener.onClickQiitaListData(qiitaArticleResponse);
            }
        });
    }

    @Override
    public int getItemCount() {
        return listData.size();
    }
}

File: app/src/main/java/com/example/intern_anrdoid_2020/view/QiitaListViewAdapter.kt
package com.example.intern_anrdoid_2020.view

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.bumptech.glide.Glide
import com.example.intern_anrdoid_2020.R
import com.example.intern_anrdoid_2020.model.response.QiitaArticleResponse
import com.example.intern_anrdoid_2020.view.QiitaListViewAdapter.QiitaListViewHolder
import kotlinx.android.synthetic.main.row_qiita_list.view.*

class QiitaListViewAdapter(private val listData: List<QiitaArticleResponse>?) : RecyclerView.Adapter<QiitaListViewHolder>() {

    lateinit var listener: OnItemClickListener

    interface OnItemClickListener {
        fun onItemClickListener(item: QiitaArticleResponse)
    }

    fun setOnItemClickListener(listener: OnItemClickListener) {
        this.listener = listener
    }

    class QiitaListViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        val ivQiitaImage: ImageView
        val tvQiitaTitle: TextView
        val tvQiitaLgtm: TextView

        init {
            ivQiitaImage = itemView.iv_user_image
            tvQiitaTitle = itemView.tv_qiita_title
            tvQiitaLgtm = itemView.tv_qiita_lgtm_counter
        }
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): QiitaListViewHolder {
        val v = LayoutInflater.from(parent.context).inflate(R.layout.row_qiita_list, parent, false)
        return QiitaListViewHolder(v)
    }

    override fun onBindViewHolder(holder: QiitaListViewHolder, position: Int) {
        listData?.let {
            val qiitaArticleResponse = listData[position]
            holder.tvQiitaTitle.text = qiitaArticleResponse.title
            holder.tvQiitaLgtm.text = qiitaArticleResponse.likesCount.toString()
            Glide.with(holder.ivQiitaImage.context).load(qiitaArticleResponse.user?.profileImageUrl).into(holder.ivQiitaImage)
            holder.itemView.setOnClickListener {
                if (position == RecyclerView.NO_POSITION) {
                    return@setOnClickListener
                }
                listener.onItemClickListener(qiitaArticleResponse)
            }
        }
    }

    override fun getItemCount(): Int {
        listData?.let {
            return listData.size
        }
        return 0
    }
}
